hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/DatanodeBlockPutter.java

```diff
@@ -0,0 +1,161 @@
+import org.apache.hadoop.hdds.scm.protocol.StorageContainerLocationProtocol;
+import org.apache.hadoop.ozone.OzoneSecurityUtil;
+
+import com.codahale.metrics.Timer;
+import org.apache.hadoop.ozone.common.Checksum;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import picocli.CommandLine.Command;
+import picocli.CommandLine.Option;
+
+/**
+ * Datanode test for block creation.
+ */
+@Command(name = "dbp",
+    aliases = "datanode-block-putter",
+    description = "Issues putBlock commands to a Ratis pipeline.  The " +
+        "blocks are associated with a list of fake chunks, which do not " +
+        "really exist.",
+    versionProvider = HddsVersionProvider.class,
+    mixinStandardHelpOptions = true,
+    showDefaultValues = true)
+public class DatanodeBlockPutter extends BaseFreonGenerator implements
+    Callable<Void> {
+
+  private static final Logger LOG =
+      LoggerFactory.getLogger(DatanodeBlockPutter.class);
+
+  @Option(names = {"-c", "--chunks-per-block"},
+      description = "Number of chunks to include in each putBlock",
+      defaultValue = "4")
+  private int chunksPerBlock;
+
+  @Option(names = {"-s", "--size"},
+      description = "Size of the fake chunks (in bytes)",
+      defaultValue = "1024")
+  private int chunkSize;
+
+  @Option(names = {"-l", "--pipeline"},
+      description = "Pipeline to use. By default the first RATIS/THREE "
+          + "pipeline will be used.",
+      defaultValue = "")
+  private String pipelineId;
+
+  private XceiverClientSpi client;
+
+  private Timer timer;
+
+  private ChecksumData checksumProtobuf;
+
+  @Override
+  public Void call() throws Exception {
+
+    init();
+
+    OzoneConfiguration ozoneConf = createOzoneConfiguration();
+    if (OzoneSecurityUtil.isSecurityEnabled(ozoneConf)) {
+      throw new IllegalArgumentException(
+          "datanode-block-putter is not supported in secure environment");
+    }
+
+    try (StorageContainerLocationProtocol scmLocationClient =
+        createStorageContainerLocationClient(ozoneConf)) {
+      Pipeline pipeline =
+          findPipelineForTest(pipelineId, scmLocationClient, LOG);
+
+      try (XceiverClientManager xceiverClientManager =
+               new XceiverClientManager(ozoneConf)) {
+        client = xceiverClientManager.acquireClient(pipeline);
+
+        timer = getMetrics().timer("put-block");
+
+        byte[] data = RandomStringUtils.randomAscii(chunkSize)
+            .getBytes(StandardCharsets.UTF_8);
+        Checksum checksum = new Checksum();
+        checksumProtobuf = checksum.computeChecksum(data).getProtoBufMessage();
+
+        runTests(this::putBlock);
+      }
+    } finally {
+      if (client != null) {
+        client.close();
+      }
+    }
+    return null;
+  }
+
+  private void putBlock(long stepNo) throws Exception {
+    ContainerProtos.DatanodeBlockID blockId =
+        ContainerProtos.DatanodeBlockID.newBuilder()
+            .setContainerID(1L)
```

Comment:
If I understand well this is intentional to use one container. This test is measuring the performance of put block over the time when the same (growing) rocksdb is used.

We can do l / bockPerContainer later to make it easier to run the test multiple times, but I can do it from the test execution side as well...